staging/src/k8s.io/apiserver/pkg/storage/errors/storage.go

```diff
@@ -30,7 +30,7 @@ func InterpretListError(err error, qualifiedResource schema.GroupResource) error
 		return errors.NewNotFound(qualifiedResource, "")
 	case storage.IsUnreachable(err):
 		return errors.NewServerTimeout(qualifiedResource, "list", 2) // TODO: make configurable or handled at a higher level
-	case storage.IsInternalError(err):
+	case storage.IsTransformerError(err) || storage.IsInternalError(err):
```

Comment:
why were these changes necessary?

Reply:
These functions translates storage layer error to API error. This PR differentiates the TransformerError from other InternalError at the storage layer, but the error returned to clients is still the 500 internal error.